### PR TITLE
Support for DuckDB Union datatype

### PR DESCRIPTION
**Pull request by gstvg**
e.g. `CREATE TABLE tbl1 (u UNION(num INTEGER, str VARCHAR));`

[https://duckdb.org/docs/sql/data_types/union.html](https://duckdb.org/docs/sql/data_types/union.html)

### CODE REVIEW

File: src/parser/mod.rs [hunk 1]
```diff
@@ -2246,6 +2246,32 @@ impl<'a> Parser<'a> {
...
+    /// UNION(field_name field_type[,...])
+    /// ```
+    ///
+    /// [1]: https://duckdb.org/docs/sql/data_types/union.html
+    fn parse_union_type_def(&mut self) -> Result<Vec<UnionField>, ParserError> {
+        self.expect_keyword(Keyword::UNION)?;
+
+        self.expect_token(&Token::LParen)?;
+
+        let fields = self.parse_comma_separated0(|p| {
```
Comment (gstvg) [inline src/parser/mod.rs:2263]: Zero field unions are invalid on DuckDB and don't even parse. Should we error here or let downstream users handle it?

[https://shell.duckdb.org/#queries=v0,create-table-tbl1-(u-union()-)~](https://shell.duckdb.org/#queries=v0,create-table-tbl1-(u-union()-)~)

Comment (jayzhan211) [reply src/parser/mod.rs:2263]: I think it is fine to return error here.

Comment (gstvg) [reply src/parser/mod.rs]: Done

**Review by iffyio — APPROVED** [review]
LGTM! cc @alamb

**Review by alamb — APPROVED** [review]
Likewise -- looks good to me -- thank you @iffyio  and @gstvg

**Pull request merged**